### PR TITLE
Changing Scene View with Offline suggestions

### PR DESCRIPTION
**Pull request by jedouard98**
- Build indexes are seen on the right.
- Right click remove selection supported.
- Enable buttons turn on and off affect build index.
- Added "Add Open Scenes" button.
- Added GUIUtility.Exit() to respond to EndLayout Errors being thrown from FilePanels 
- Added Textbox and browse support for assetbundle creation name
- Added TODO for drag and drop

Demo: 

ezgif com-video-to-gif

### CODE REVIEW

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs [hunk 1]
```diff
@@ -42,32 +44,67 @@ public PlayInstantSceneTreeView(TreeViewState treeViewState)
         public class SceneItem : TreeViewItem
         {
             public bool Enabled;
+            public bool OldEnabledValue;
+            public string SceneBuildIndexString;
+            
+            public override bool Equals(object obj)
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs:51]:
```
            public override bool Equals(object obj)
            {
                var that = obj as SceneItem;
                 return that != null && this.displayName.Equals(that.displayName);
            }
```

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs [hunk 2]
```diff
@@ -42,32 +44,67 @@ public PlayInstantSceneTreeView(TreeViewState treeViewState)
...
             }
 
-            SetupParentsAndChildrenFromDepths(root, allItems);
+            EditSceneBuildIndexString();
+            Reload();
+        }
+
+        private void EditSceneBuildIndexString()
+        {
+            var i = 0;
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs]: Rename to "sceneId" or something clearer

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs [hunk 3]
```diff
@@ -42,32 +44,67 @@ public PlayInstantSceneTreeView(TreeViewState treeViewState)
...
+            foreach (var item in _allItems)
+            {
+                var sceneItem = (SceneItem) item;
+                sceneItem.id = i;
+                sceneItem.SceneBuildIndexString = sceneItem.Enabled ? "" + i++ : "";
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs]: This is strange. You're only incrementing the ID if a scene is enabled, but on the previous line you're always setting the ID. So in this case with 3 scenes:
enabled
disabled
enabled

you'd end up with sceneItem.id as follows:
enabled 0
disabled 1
enabled 1

Please revisit.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 4]
```diff
@@ -46,6 +46,8 @@ public enum ToolBarSelectedButton
 
         private const int WindowMinWidth = 475;
         private const int WindowMinHeight = 400;
+        
+        private const int SceneViewDistFromTop = 230;
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: s/Dist/Distance/ or maybe "Delta"

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 5]
```diff
@@ -148,16 +150,41 @@ private void OnGuiCreateBundleSelect()
...
+            QuickDeployConfig.AssetBundleFileName = EditorGUILayout.TextField(QuickDeployConfig.AssetBundleFileName,
+                GUILayout.MinWidth(FieldMinWidth));
+            EditorGUILayout.EndHorizontal();
 
+            EditorGUILayout.BeginHorizontal();
+            GUILayout.FlexibleSpace();
+            if (GUILayout.Button("Browse", GUILayout.Width(ShortButtonWidth)))
+            {
+                QuickDeployConfig.AssetBundleFileName = EditorUtility.SaveFilePanel("Save AssetBundle", "", "", "");
+                GUIUtility.ExitGUI();
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Add a comment saying what this line is needed for.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 6]
```diff
@@ -240,9 +267,11 @@ private void OnGuiDeployBundleSelect()
             {
                 QuickDeployConfig.CloudCredentialsFileName =
                     EditorUtility.OpenFilePanel("Select cloud credentials file", "", "");
+                GUIUtility.ExitGUI();
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: same

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 7]
```diff
@@ -255,9 +284,10 @@ private void OnGuiDeployBundleSelect()
             if (GUILayout.Button("Browse", GUILayout.Width(ShortButtonWidth)))
             {
                 QuickDeployConfig.AssetBundleFileName = EditorUtility.OpenFilePanel("Select AssetBundle file", "", "");
+                GUIUtility.ExitGUI();
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: same

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 8]
```diff
@@ -356,6 +386,7 @@ private void OnGuiLoadingScreenSelect()
             {
                 _loadingScreenImagePath =
                     EditorUtility.OpenFilePanel("Select Image", "", "png,jpg,jpeg,tif,tiff,gif,bmp");
+                GUIUtility.ExitGUI();
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: same

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs [hunk 9]
```diff
@@ -403,6 +434,7 @@ private void OnGuiCreateBuildSelect()
             {
                 QuickDeployConfig.ApkFileName =
                     EditorUtility.SaveFilePanel("Choose file name and location", "", "base.apk", "apk");
+                GUIUtility.ExitGUI();
```
Comment (thanwah) [inline GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: same

**Comment by thanwah** [comment]
By the way, that gif demo is awesome! Thanks for providing that!

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 4 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs (hunk 1 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs:51]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs (hunk 2 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 7 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 9 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 8 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 6 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs (hunk 5 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/QuickDeployWindow.cs]: Resolved.

File: GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs (hunk 3 above)
Comment (jedouard98) [reply GooglePlayInstant/Editor/QuickDeploy/PlayInstantSceneTreeView.cs]: Thanks for this issue. I've reproduced this and it's exactly as you said. I don't know why I was initially changing the sceneItem id to begin with. Thanks!

**Review by thanwah — APPROVED** [review]
LGTM, thanks!

**Pull request merged**